prow/cmd/deck/main_test.go

```diff
@@ -1050,3 +1051,115 @@ func TestCanTriggerJob(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateStoragePath(t *testing.T) {
+	boolTrue := true
+	boolFalse := false
+	testCases := []struct {
+		name        string
+		config      config.ProwConfig
+		path        string
+		expectedErr string
+		httpStatus  int
+	}{
+		{
+			name:        "unspecified config means no validation",
+			config:      config.ProwConfig{Deck: config.Deck{}},
+			path:        "to/some/wild/location",
+			expectedErr: "",
+		},
+		{
+			name:        "validation disabled",
+			config:      config.ProwConfig{Deck: config.Deck{RestrictStoragePaths: &boolFalse}},
+			path:        "to/some/wild/location",
+			expectedErr: "",
+		},
+		{
+			name:        "validation enabled",
+			config:      config.ProwConfig{Deck: config.Deck{RestrictStoragePaths: &boolTrue}},
+			path:        "to/some/wild/location",
+			expectedErr: "bucket \"some\" not in allowed list",
+			httpStatus:  http.StatusBadRequest,
+		},
+		{
+			name:        "validation enabled, bad path format",
+			config:      config.ProwConfig{Deck: config.Deck{RestrictStoragePaths: &boolTrue}},
+			path:        "doesn't look like a properly formatted storage path! :(",
+			expectedErr: "invalid path: doesn't look like a properly formatted storage path",
+			httpStatus:  http.StatusBadRequest,
+		},
+		{
+			name: "DecorationConfig allowed bucket",
+			config: config.ProwConfig{
+				Deck: config.Deck{RestrictStoragePaths: &boolTrue},
+				Plank: config.Plank{
+					DefaultDecorationConfigs: map[string]*prowapi.DecorationConfig{
+						"*": {
+							GCSConfiguration: &prowapi.GCSConfiguration{
+								Bucket: "kubernetes-jenkins",
+							},
+						},
+					},
+				},
+			},
+			path:        "gs/kubernetes-jenkins/pr-logs/directory/pull-capi",
+			expectedErr: "",
+		},
+		{
+			name:        "custom allowed bucket",
+			config:      config.ProwConfig{Deck: config.Deck{RestrictStoragePaths: &boolTrue, AdditionalAllowedBuckets: []string{"kubernetes-prow"}}},
+			path:        "gs/kubernetes-prow/random-folder/crazy-subfolder",
+			expectedErr: "",
+		},
+		{
+			name:        "unknown bucket path",
+			config:      config.ProwConfig{Deck: config.Deck{RestrictStoragePaths: &boolTrue}},
+			path:        "gs/istio-prow/pr-logs/directory/post-check",
+			expectedErr: "bucket \"istio-prow\" not in allowed list",
+			httpStatus:  http.StatusBadRequest,
+		},
+	}
+
+	for _, tc := range testCases {
+		cfg := config.Config{ProwConfig: tc.config}
+		ca := config.Agent{}
+		ca.Set(&cfg)
+		expectingErr := len(tc.expectedErr) > 0
```

Comment:
While not technically incorrect its confusing to have part of the testcase in the main test and part in ths sub test, could you move all of it into the func that is called below with `t.Run`?

Reply:
Done.